src/java.base/share/classes/sun/security/ssl/SSLSessionContextImpl.java

```diff
@@ -170,6 +178,52 @@ public int getSessionCacheSize() {
         return cacheLimit;
     }
 
+    // Package-private, used only from SSLContextImpl::engineInit() to initialie currentKeyID.
+    void initCurrentKeyID(int keyID) {
```

Comment:
The ID will work as if it is unique in the context, granted with synchronization.  It may be not necessary to use secure number for it.   The ID could be assigned at constructor (see comment above, use number zero or the current time, etc.), and thus this method and the caller code could be removed, I think.

Reply:
I'm not an expert in this area, but [RFC 5077 reccomends](https://www.rfc-editor.org/rfc/rfc5077#page-10) that the ID "*should be randomly generated to avoid collisions between servers*". What about initializing it to the `hashCode()` of the `SSLSessionContextImpl` object?